src/Compilers/CSharp/Portable/CodeGen/Optimizer.cs

```diff
@@ -1239,21 +1243,23 @@ public override BoundNode VisitConditionalGoto(BoundConditionalGoto node)
         public override BoundNode VisitConditionalOperator(BoundConditionalOperator node)
         {
             var origStack = StackDepth();
-            BoundExpression condition = (BoundExpression)this.Visit(node.Condition);
+            BoundExpression condition = (BoundExpression)this.VisitExpression(node.Condition, ExprContext.Value);
```

Comment:
`(BoundExpression)` cast is not needed, here nor in other `VisitExpression` calls below.